scene/2d/camera_2d.cpp

```diff
@@ -77,7 +77,18 @@ void Camera2D::_update_process_mode() {
 }
 
 void Camera2D::set_zoom(const Vector2 &p_zoom) {
-	zoom = p_zoom;
+	Vector2 new_zoom = p_zoom;
+
+	// Setting zoom to zero causes 'affine_invert' issues
+	if (Math::is_zero_approx(p_zoom.x)) {
+		new_zoom.x = CMP_EPSILON * SGN(p_zoom.x);
+	}
+
+	if (Math::is_zero_approx(p_zoom.y)) {
+		new_zoom.y = CMP_EPSILON * SGN(p_zoom.y);
+	}
```

Comment:
Not sure, but wouldn't clamping be cleaner and have less operations?

Something like:

```
zoom.x = max(abs(p_zoom.x), CMP_EPSILON) * sign(p_zoom.x)
zoom.y = max(abs(p_zoom.y), CMP_EPSILON) * sign(p_zoom.y)
```

Reply:
I wouldn't recommend to do it that way - because it's allowed to have negative values for a zoom as well. It's just problematic when scale is around zero.

Reply:
It works with negatives.  Did I miss something?